src/types.ts

```diff
@@ -0,0 +1,27 @@
+import type { Readable } from 'node:stream';
+import type { Dispatcher, FormData } from 'undici'
+import type { IncomingHttpHeaders } from 'undici/types/header.js';
+import type BodyReadable from 'undici/types/readable.js';
+
+export type RequestOptions = { // partially derived from undici.Dispatcher.RequestOptions
+    url: string;
+    method: Dispatcher.HttpMethod;
+    /** 
+     * @default null 
+     * */
+    body?: string | Buffer | Uint8Array | Readable | null | FormData;
+    /**
+     * Headers to be sent with the request.
+     * 
+     * @default new Headers()
+     */
+    headers?: Headers;
+}
+
+export type ResponseData = { // partially derived from undici.Dispatcher.ResponseData
+    statusCode: number;
+    headers: IncomingHttpHeaders;
+    body: BodyReadable.default;
```

Comment:
```suggestion
    body: Dispatcher.ResponseData['body'];
```

Same general idea as the line above - the one difference in this type access is that it would give you undici's mixin combined with BodyReadable.default, but I think that's accurate.